### PR TITLE
AEA-1573 fix minor issues on MultiAgentManager

### LINKED ISSUE #2482: Examples of unhelpful error messages
**Is your feature request related to a problem? Please describe.**
This issue lists unhelpful error messages so they can be made more informative.

**Describe the solution you'd like**
Make them informative and actionable

Screenshot 2021-04-29 at 14 15 51

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

- add check 'ensure_private_keys' at the end of AgentAlias.set_overrides, which in turn is called by 'MultiAgentManager.add_agent'.
- Print a warning message when an exception occurred but no error callback is registered.

## Fixes

Partially fixes #2482

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/manager/manager.py [hunk 1]
```diff
@@ -263,8 +267,11 @@ async def _manager_loop(self) -> None:
                 agent_name = agents_run_tasks_futures[task]
                 self._agents_tasks.pop(agent_name)
                 if task.exception():
-                    for callback in self._error_callbacks:
-                        callback(agent_name, task.exception())
+                    if len(self._error_callbacks) == 0:
+                        self._print_exception_occurred_but_no_error_callback(agent_name)
```
Comment (solarw) [inline aea/manager/manager.py:271]: make it default callback?

**Review by solarw — APPROVED** [review]
LGTM

File: aea/manager/manager.py (hunk 1 above)
Comment (DavidMinarsch) [inline aea/manager/manager.py:271]: why not print the exception too? I.e. pass task.exception() here?

File: aea/manager/manager.py [hunk 2]
```diff
@@ -843,3 +850,20 @@ def _save_state(self) -> None:
...
+        """
+        Print a warning message when an exception occurred but no error callback is registered.
+
+        :param agent_name: the agent name.
+        :return: None
+        """
+        if self._warning_message_printed_for_agent.get(agent_name, False):
+            return
+        self._warning_message_printed_for_agent[agent_name] = True
+        self._print_exception_occurred_but_no_error_callback(agent_name)
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:864]: recursion?

File: aea/manager/project.py [hunk 3]
```diff
@@ -301,7 +301,8 @@ def set_overrides(
                 )
 
         overrides["component_configurations"] = component_configurations
-        return self.agent_config_manager.update_config(overrides)
+        self.agent_config_manager.update_config(overrides)
+        self._ensure_private_keys()
```
Comment (DavidMinarsch) [inline aea/manager/project.py:305]: Not sure if this makes sense here; I'd prefer it to be only called if agent_overrides are not None; to make explicit fact that it depends on changes in agent overrides

File: aea/manager/manager.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/manager/manager.py:271]: Yes that's a good idea. Why don't we have a default callback?

Comment (marcofavorito) [reply aea/manager/manager.py:271]: Then it means we want a default callback, not a warning message

Comment (marcofavorito) [reply aea/manager/manager.py:271]: if someone adds a callback with `add_error_callback`, should the default callback be replaced?

Comment (DavidMinarsch) [reply aea/manager/manager.py:271]: Yes

Comment (DavidMinarsch) [reply aea/manager/manager.py:271]: Yeah I guess, it would be better for development

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**